retry/integration_test.go

```diff
@@ -180,10 +185,31 @@ func (s *RetryTripperwareSuite) TestCustomResponseDiscarder() {
 			http_retry.WithBackoff(http_retry.BackoffLinear(retryTimeout)),
 		),
 	}
-	req := s.createRequest("GET", s.SimpleCtx())
+	req := s.createRequest("GET", s.SimpleCtx(), expectedContent)
 	resp, err := s.NewClient().Do(req)
 	require.NoError(s.T(), err, "call shouldn't fail")
 	assert.Equal(s.T(), failureCode, resp.StatusCode, "failure code should be propagated")
 	require.EqualValues(s.T(), 5, s.f.requestCount(), "backend should see all the retry calls")
 	s.ClientTripperware = clientTripperware
 }
+
+func (s *RetryTripperwareSuite) TestRequestBodyBuffering() {
+	s.f.resetFailingConfiguration(3, noSleep)
+	req := s.createRequest("GET", s.SimpleCtx(), expectedContent)
+	removeGetBody(req)
+	resp, err := s.NewClient().Do(req)
+	require.NoError(s.T(), err, "call shouldn't fail")
+	require.Equal(s.T(), httpwares_testing.DefaultPingBackStatusCode, resp.StatusCode, "response should succeed")
+	require.EqualValues(s.T(), 3, s.f.requestCount(), "3 requests should be retried to meet the modulo")
+}
+
+func (s *RetryTripperwareSuite) TestRequestBodyBufferingWithNoBody() {
+	s.f.resetFailingConfiguration(3, noSleep)
+	expectedContent = ""
```

Comment:
Use a local variable to store and reset this value, eliminate the extra package variable.